### PR TITLE
Implements one click eris startup and shutdown of services (Issue #766)

### LINKED ISSUE #766: [Core] One click shutdown/startup for dev machines
Can we have a one click roll up/down? Like `eris startup` `eris shutdown`? It would be handy for dev machines where you want to end a marmoty day of dev and gracefully shutdown everything and return the next day in the same state. I think that it would just wrap `eris services stop xxxxx` and `eris chains stop xxxx` and vice-versa against all running containers.

### PR DESCRIPTION
**Pull request by slowtokyo**
When submitting a pull request (branched from develop):

- [ check ] `gofmt -s` your code
- [ check ] `git rebase develop` while on your feature/fix branch
- [ check ] use a descriptive title
- [ closes #766 ] in the comments section, include the issue # (i.e., closes #577)
- [ Adds the ability to `eris startup` which starts up all existing services that you have. `startup` also takes a `--chain` flag to simultaneously startup a chain if you like. Adds ability to `eris shutdown` which gracefully shuts down all chains and services currently running. Adds function to `KillChains` which is the same functionality as the `KillChain` command except it takes an array of chain container names in case multiple chains are running. This is similar to the behaviour of `KillServices`. ] provide a significant summary of the important changes the PR yields. Describe under-the-hood behaviour of code changes and/or modifications to the user experience
- [ ] await a thumbs up (:+1:) from the maintainers of the repository

### CODE REVIEW

**Comment by zramsay** [comment]
Thanks for the PR @slowtokyo :)
Unfortunately, this will have to wait until after the current release cycle - the codebase around chains (and everything else) is too much in flux at the moment. 
- [ ] Note also that we'll want to add in the tests I wrote back when we started. (https://github.com/eris-ltd/eris-cli/commit/1a6ddf4181999edd374b9dcc176a164d06dc826e)

File: chains/operate.go [hunk 1]
```diff
@@ -43,6 +43,46 @@ func InstallChain(do *definitions.Do) error {
 	return setupChain(do, loaders.ErisChainInstall)
 }
 
+//function for killing multiple chains (takes array)
+func KillChains(do *definitions.Do) error {
```
Comment (zramsay) [inline chains/operate.go]: I'm not sure we want an additional function here.

File: cmd/start_stop.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package commands
+
+import (
+	//"github.com/eris-ltd/eris-cli/definitions"
```
Comment (zramsay) [inline cmd/start_stop.go]: this line should be completely removed

File: cmd/start_stop.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/eris-ltd/eris-cli/kaihei"
```
Comment (zramsay) [inline cmd/start_stop.go]: was a good name as a placeholder but we'll want something more relevant.

File: definitions/chains.go [hunk 4]
```diff
@@ -7,6 +7,7 @@ type Chain struct {
 	ChainID string `mapstructure:"chain_id" json:"chain_id" yaml:"chain_id" toml:"chain_id"`
 	// type of the chain
 	ChainType string `mapstructure:"chain_type" json:"chain_type" yaml:"chain_type" toml:"chain_type"`
+	Chain string `json:"chain,omitempty" yaml:"chain,omitempty" toml:"chain,omitempty"`
```
Comment (zramsay) [inline definitions/chains.go]: can we get by without an additional variable?

File: kaihei/start.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	"github.com/eris-ltd/eris-cli/util"
+	srv "github.com/eris-ltd/eris-cli/services"
+	"github.com/eris-ltd/eris-cli/definitions"
+)
+
+func StartUpEris(do *definitions.Do) error {
+
+	// start chain
+	// doChain.Name - name of the chain (optional)
+	if do.ChainName != ""{
```
Comment (zramsay) [inline kaihei/start.go]: gofmt

File: kaihei/start.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+		doChain := definitions.NowDo()
+		doChain.Name = do.ChainName
+
+		fmt.Println("starting up your chain...")
+		if err := chains.StartChain(doChain); err != nil {
+			return err
+		}
+	}
+	
+	fmt.Println("starting up your services...")
```
Comment (zramsay) [inline kaihei/start.go]: `log.Warn()`

File: kaihei/start.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	if len(listOfServices) == 0 {
+		return fmt.Errorf("no existing services to start")
+	}
+
+	names := make([]string, len(listOfServices))
+	for i, serviceName := range listOfServices {
+		names[i] = serviceName.ShortName
+	}
+
+	fmt.Println(names)
```
Comment (zramsay) [inline kaihei/start.go]: remove?

File: kaihei/start.go [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+	doStart := definitions.NowDo()
+	doStart.ServicesSlice = names
+	if err := srv.StartService(doStart); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ShutUpEris(do *definitions.Do) error {
```
Comment (zramsay) [inline kaihei/start.go]: name choice?

File: kaihei/start.go [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+	fmt.Println("shutting down your services...")
```
Comment (zramsay) [inline kaihei/start.go]: log.Warn()

File: kaihei/start.go [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+	if len(listOfServices) == 0 {
+		return fmt.Errorf("no existing services to stop")
+	}
+
+	names := make([]string, len(listOfServices))
+	for i, serviceName := range listOfServices {
+		names[i] = serviceName.ShortName
+	}
+
+	fmt.Println(names)
```
Comment (zramsay) [inline kaihei/start.go]: remove?

File: services/operate.go [hunk 11]
```diff
@@ -79,7 +79,7 @@ func KillService(do *definitions.Do) (err error) {
 		}
 		services = append(services, s...)
 	}
-
+
```
Comment (zramsay) [inline services/operate.go]: gofmt

File: chains/operate.go (hunk 1 above)
Comment (slowtokyo) [inline chains/operate.go]: Yah, I wasn't sure whether or not we wanted that. I wasn't sure whether that was a dependency for other functions so created a new one instead. The question that needs answering is: _Are there cases where you would want to only stop one chain and leave others running or do you want `eris chains stop` to stop all chains all the time?_

File: definitions/chains.go (hunk 4 above)
Comment (slowtokyo) [inline definitions/chains.go]: Yes, I think so. Meant to delete that.

File: kaihei/start.go (hunk 7 above)
Comment (slowtokyo) [inline kaihei/start.go]: Sorry, remove the whole 15 lines or so?

File: kaihei/start.go (hunk 8 above)
Comment (slowtokyo) [inline kaihei/start.go]: haha, that was your bag man. Can think of something else. I like `ShutUpEris` though

File: kaihei/start.go (hunk 10 above)
Comment (slowtokyo) [inline kaihei/start.go]: remove what?

File: kaihei/start.go (hunk 7 above)
Comment (zramsay) [inline kaihei/start.go]: line 40. if we want logging, then see above note about the `eris-logger` package

Comment (zramsay) [inline kaihei/start.go]: (`fmt.Println()` or `Printf()` is great for debugging, however....i sometimes leave these lying around as well.)

File: kaihei/start.go (hunk 8 above)
Comment (zramsay) [inline kaihei/start.go]: `ShutDownEris` maybe. it's not like eris is being noisy just by having a chain and some services running ;)

File: kaihei/start.go (hunk 10 above)
Comment (zramsay) [inline kaihei/start.go]: fmt.Println. Or use `eris-logger` alongside the `WithFields` parameter, if appropriate

**Comment by zramsay** [comment]
@slowtokyo thanks again for the PR. Unfortunately, we'll be closing for now as we are moving in the direction of "trimming the fat" and reducing the number of commands for eris-cli. I would recommend sticking with the [two commands here](https://github.com/eris-ltd/eris-cli/issues/766#issuecomment-230478836) in a custom startup/shutdown script.

**Pull request closed without merging**